src/lib/y2network/interface_config_builder.rb

```diff
@@ -43,6 +50,27 @@ def [](key)
       @config[key]
     end
 
+    # how many device names is proposed
+    NEW_DEVICES_COUNT = 10
+    # Propose bunch of possible names for interface
+    # do not modify anything
+    # @return [Array<String>]
+    def proposed_names
+      Yast::LanItems.new_type_devices(@settings.type, NEW_DEVICES_COUNT)
+    end
+
+    def valid_name?(name)
+      !(value != /^[[:alnum:]._:-]{1,15}\z/)
+    end
+
+    def already_exists?(name)
+      Yast::NetworkInterfaces.List("").include?(name)
+    end
+
+    def name_valid_characters
+      Yast::NetworkInterfaces.ValidCharsIfcfg
+    end
```

Comment:
seeing all those method, maybe having object for name makes sense?

Reply:
I think that this method is not correct. It is used only at one place for validating ifcfg name but it can validate even characters which are not allowed in ifcfg (respectively network device) names - see https://github.com/yast/yast-yast2/blob/master/library/types/src/modules/String.rb#L591 and https://github.com/yast/yast-yast2/blob/master/library/types/src/modules/String.rb#L42
Correct validator is a few lines above - valid_name?

Reply:
so is it bug in ValidCharsIfcfg? I just need to get array with chars that is allowed.

Reply:
i think so